custom_metrics/almanac.js

```diff
@@ -306,6 +359,240 @@ return JSON.stringify({
+      if (count > 1) {
+        // Subtract one because the first div to have this ID is not a duplicate
+        total_duplicates += count - 1;
+      }
+    }
+
+    return total_duplicates;
+  })(),
+
+  /**
+   * The 'h' is stripped to create a numeric array.
+   * E.g. h1 > h2 > h3 > h3 => [1, 2, 3, 3, ]
+   */
+  'headings_order': (() => {
+    const headings = [...document.querySelectorAll('h1, h2, h3, h4, h5, h6')];
+    const levels = [];
+    for (const heading of headings) {
+      const level = parseInt(heading.tagName.replace('H', ''), 10);
+      if (!isNaN(level)) {
+        levels.push(level);
+      }
+    }
+
+    return levels;
+  })(),
+
+  'shortcuts_stats': (() => {
+    const aria_shortcut_nodes = [...document.querySelectorAll('[aria-keyshortcuts]')];
+    const accesskey_nodes = [...document.querySelectorAll('[accesskey]')];
+
+    return {
+      total_with_aria_shortcut: aria_shortcut_nodes.length,
+      total_with_accesskey: accesskey_nodes.length,
+
+      // Purposely left these as potentially duplicated fields so we can analyze if the same value is used more than once
+      aria_shortcut_values: aria_shortcut_nodes.map(node => node.getAttribute('aria-keyshortcuts')),
+      accesskey_values: accesskey_nodes.map(node => node.getAttribute('accesskey')),
+    };
+  })(),
+
+  'nodes_using_aria': (() => {
+    // Example usage: Process all elements on the page
+    const aria_nodes = [];
+    walkNodes(document.documentElement, (node) => {
+      const attributes = node.getAttributeNames();
+      if (attributes.length <= 0) {
+        return;
+      }
+
+      let has_aria = false;
+      for (const attribute_name of attributes) {
+        if (attribute_name.toLowerCase().indexOf('aria-') === 0) {
+          // This node has aria, so we'll store all of its attributes and move on to the next node now
+          aria_nodes.push(parseNode(node));
+          return;
+        }
+      }
+
+      // The node didn't have aria so we simply do nothing and move on to the next node
+    });
+
+    return aria_nodes;
+  })(),
+
+  // NOTE: This will not pick up all of the attributes on scripts
+  'attributes_used_on_elements': (() => {
+    // Example usage: Process all elements on the page
+    const unique_values = new Set();
+    const attributes_and_count = {};
+    walkNodes(document.documentElement, (node) => {
+      const attribute_names = node.getAttributeNames();
+      if (attribute_names.length <= 0) {
+        return;
+      }
+
+      // Count how often each of these attributes shows up
+      for (const name of attribute_names) {
+        unique_values.add(name);
+
+        if (!attributes_and_count[name]) {
+          attributes_and_count[name] = 1;
+          continue;
+        }
+
+        attributes_and_count[name]++;
+      }
+    });
+
+    return {
+      // How often each of these values show up
+      values_and_count: attributes_and_count,
+      unique_values: [...unique_values],
```

Comment:
Same comment as `nodes_using_role`. Do we need both?

Reply:
The Markup chapter wants this. All values an their counts.

Reply:
Keeping values_and_count in favor of unique_values. This function will now return the `values_and_count` object itself though, as there is nothing else to return